### ISSUE TITLE
Class Wiringa\DumpServer\Console\ServerDumpCommand\Interceptor does not exist

### ISSUE THREAD

**Issue by euperia** [DuckThom/magento2-dump-server#4]
I'm upgrading from Magento 2.4.2 to 2.4.3 and get the following whenever I try enabling the module:

```
/var/www/html$ php bin/magento module:enable Wiringa_DumpServer
No modules were changed.

In ClassReader.php line 24:

  Class Wiringa\DumpServer\Console\ServerDumpCommand\Interceptor does not exist
```

Running on Ubuntu 20.04, PHP 7.4.3.

**Comment by rofthedeep** [comment]
Same here - did you find a solution?

**Comment by DuckThom** [comment]
I'm unable to reproduce the issue. 
Can you check if dev mode is enabled? Else, try running `bin/magento setup:di:compile` to make it generate the interceptors and see if that fixes the issue.

**Comment by rofthedeep** [comment]
@DuckThom thx for your answer!
I found the problem: my symfony/var-dumper version was too new (5.3.8).
It works with a requirement for "symfony/var-dumper": "^v4.4" in the require-dev section. Perhaps you can add it here in the module?

The following error was thrown:
`
[2022-02-28 12:55:09] main.CRITICAL: Class Symfony\Component\Console\Completion\CompletionInput does not exist
Class Wiringa\DumpServer\Console\ServerDumpCommand\Interceptor generation error: The requested class did not generate properly, because the 'generated' directory permission is read-only. If --- after running the 'bin/magento setup:di:compile' CLI command when the 'generated' directory permission is set to write --- the requested class did not generate properly, then you must add the generated class object to the signature of the related construct method, only. [] []
`

Best, Tim

**Comment by DuckThom** [comment]
@rofthedeep Thanks for investigating, though I'm a bit curious as to why that error is coming from the dump command. To me it looks like there is some conflict/mismatch between some installed symfony components. 

The `Symfony\Component\VarDumper\Command\ServerDumpCommand::complete` method, added in `symfony/var-dumper` 5.4, is the one that requires a parameter of the type `Symfony\Component\Console\Completion\CompletionInput`. And this class was added in and is only used by `symfony/console` 5.4.

Could you check your version of `symfony/console` (`composer show symfony/console`)? Because I don't think this is caused by my module since I haven't been able to reproduce this on multiple projects, all of which have `symfony/console` 4.4.38 and `symfony/var-dumper` 4.4, 5.3 or 5.4 installed, which is matching the version constraint set by Magento https://github.com/magento/magento2/blob/2.4-develop/composer.json#L79 

I would prefer not to remove symfony/var-dumper ^5.0 from this module because as far as I can tell, it is (and should be) compatible with that constraint. Though I can add a constraint for symfony/console 4.4 if that turns out to be the culprit.

**Comment by rofthedeep** [comment]
Hi @DuckThom,

thx for your message!

Here is the result of the command:

```
name     : symfony/console
descrip. : Eases the creation of beautiful and testable command line interfaces
keywords :
versions : * v4.4.37
type     : library
license  : MIT License (MIT) (OSI approved) https://spdx.org/licenses/MIT.html#licenseText
homepage : https://symfony.com
source   : [git] https://github.com/symfony/console.git 0259f01dbf9d77badddbbf4c2abb681f24c9cac6
dist     : [zip] https://api.github.com/repos/symfony/console/zipball/0259f01dbf9d77badddbbf4c2abb681f24c9cac6 0259f01dbf9d77badddbbf4c2abb681f24c9cac6
path     : /Volumes/Development/_rocket-media/dtack/dtack-magento2/www/vendor/symfony/console
names    : symfony/console, psr/log-implementation

support
source : https://github.com/symfony/console/tree/v4.4.37

autoload
psr-4
Symfony\Component\Console\ => .
exclude-from-classmap

requires
php >=7.1.3
symfony/polyfill-mbstring ~1.0
symfony/polyfill-php73 ^1.8
symfony/polyfill-php80 ^1.16
symfony/service-contracts ^1.1|^2

requires (dev)
psr/log ^1|^2
symfony/config ^3.4|^4.0|^5.0
symfony/dependency-injection ^3.4|^4.0|^5.0
symfony/event-dispatcher ^4.3
symfony/lock ^4.4|^5.0
symfony/process ^3.4|^4.0|^5.0
symfony/var-dumper ^4.3|^5.0

suggests
psr/log For using the console logger
symfony/event-dispatcher
symfony/lock
symfony/process

provides
psr/log-implementation 1.0|2.0

conflicts
psr/log >=3
symfony/dependency-injection <3.4
symfony/event-dispatcher <4.3|>=5
symfony/lock <4.4
symfony/process <3.3
```


I already set the requirement for the ```symfony/var-dumper``` in our project on ```^v4.4``` - thus does this information help?

**Comment by DuckThom** [comment]
@rofthedeep I've managed to reproduce it and pushed a fix in version `1.1.1`.

The `symfony/var-dumper` version is now limited to max 5.3.* (the version before the `complete` method was added) which has fixed the issue for me.

I've also added PHP 8.1 so this should also support Magento 2.4.4 now
